### PR TITLE
Softcoded achievements

### PR DESCRIPTION
**Pull request by tposejank**
example of an achievement

```
{
    "name": "Name here",
    "icon": "image name here",
    "description": "Description ",
    "unlocksAfter": "the week it unlocks after",
    "hidden": (is it hidden? true or false)
    "customGoal": (is this goal customizable? true or false)
}
```

for custom goals, when the game checks for achievements it calls function `onCheckForAchievement` with arguments `name`
do return Function_Continue to notify the game it is unlockable
example:

`if getProperty('misses') > 10 and leName == 'lmao_skill_issue' then return Function_Continue end`

on the luas.

### CODE REVIEW

**Comment by fnaffan29** [comment]
WE NEED THIS TO GET APPROVED

**Comment by ethannn123** [comment]
> WE NEED THIS TO GET APPROVED

I AGREE

**Comment by XtraXD1** [comment]
> > WE NEED THIS TO GET APPROVED
> 
> I AGREE

x3

**Comment by skedgyedgy** [comment]
-- yoooooooooo

**Comment by MorenoTheCappuccinoChugger** [comment]
WE NEED THIS TO GET APPROVED
 I AGREE
 x3
AND FOR FREE

**Comment by MrClogsworthYT** [comment]
@ShadowMario I swear if you don't merge

**Comment by supajimmy** [comment]
Yo we need this!

**Comment by bbpanzu** [comment]
integrate with lua and i may pull < 33

**Comment by tposejank** [comment]
> integrate with lua and i may pull < 33

what

**Comment by tposejank** [comment]
I don't know, this..??

**Comment by tposejank** [comment]
@bbpanzu

File: source/PlayState.hx [hunk 1]
```diff
@@ -4574,11 +4594,11 @@ class PlayState extends MusicBeatState
 							unlock = true;
 						}
 				}
+			}
 
-				if(unlock) {
-					Achievements.unlockAchievement(achievementName);
-					return achievementName;
-				}
```
Comment (DaviGamerDias12) [inline source/PlayState.hx:4581]: -221

Comment (skedgyedgy) [reply source/PlayState.hx:4581]: -- literally immediately below it
image

File: source/AchievementsMenuState.hx [hunk 2]
```diff
@@ -42,7 +42,7 @@ class AchievementsMenuState extends MusicBeatState
 		add(grpOptions);
 
 		for (i in 0...Achievements.achievementsStuff.length) {
-			if(!Achievements.achievementsStuff[i][3] || Achievements.achievementsMap.exists(Achievements.achievementsStuff[i][2])) {
+			if(!Achievements.achievementsStuff[i][4] || Achievements.achievementsMap.exists(Achievements.achievementsStuff[i][2])) {
```
Comment (Wither362) [inline source/AchievementsMenuState.hx:45]: you need a contains thing for the duplication bug

**Pull request merged**